packages/@lwc/engine-core/src/framework/wiring.ts

```diff
@@ -16,6 +16,36 @@ const DeprecatedWiredParamsMeta = '$$DeprecatedWiredParamsMetaKey$$';
 const WireMetaMap: Map<PropertyDescriptor, WireDef> = new Map();
 function noop(): void {}
 
+interface WireContextInternalProtocolPayload {
+    setNewContext(newContext: ContextValue): void;
+    setDisconnectedCallback(disconnectCallback: () => void): void;
+}
+
+export interface WireContextRegistrationEvent
+    extends CustomEvent<undefined>,
+        WireContextInternalProtocolPayload {}
+
+function createWireContextRegistrationEvent(
+    adapterContextToken: string,
+    { setNewContext, setDisconnectedCallback }: WireContextInternalProtocolPayload
+): WireContextRegistrationEvent {
+    const event = new CustomEvent<undefined>(adapterContextToken, {
```

Comment:
why not a class with a constructor doing the expandos?

Reply:
great question (i was about to ask for guidance when i saw your question). i tried to do that, but typescript does not handle it well when is done as an expando (the class should be used in `context-provider.ts`), and i don't know how to achieve the non-configurable, etc, without using decorators (`experimentalDecorators` flag). maybe @pmdartus have some advise here.

alternatively, we can declare a readonly getter, but that can be overridden.

any ideas on how to improve this is welcomed. Ex:
```js
export class WireContextRegistrationEvent extends CustomEvent<undefined> {
    // without this 2, instances of WireContextRegistrationEvent, don't have setNewContext and setDisconnectedCallback.
    public readonly setNewContext: (newContext: ContextValue) => void;
    public readonly setDisconnectedCallback: (disconnectCallback: () => void) => void;

    constructor(
        adapterToken: string, 
        { setNewContext, setDisconnectedCallback }: WireContextInternalProtocolPayload
    ) {
        super(adapterToken, {
            bubbles: true,
            composed: true
        });

        // Without this 2 lines, ts shows 2 errors:
        // TS2564: Property 'setDisconnectedCallback' has no initializer and is not definitely assigned in the constructor.
        // this.setNewContext = setNewContext;
        // this.setDisconnectedCallback = setDisconnectedCallback;

        // and without these, setNewContext and setDisconnectedCallback are configurable, writable.
        defineProperties(this, {
            setNewContext: {
                value: setNewContext,
            },
            setDisconnectedCallback: {
                value: setDisconnectedCallback,
            },
        });
    }
}
```

Personally, i like the class variant, but for it to be possible, one of this 2 should happen:
- (preferred) we find a way of making `setNewContext` and `setDisconnectedCallback` non configurable and non writable.
- relax the constrains and allow `setNewContext` and `setDisconnectedCallback` to be configurable, writable etc. imo, this is good enough for this case.

Reply:
@caridy alternatively, we can disable the ts error, check f04485d